client/repo.c

```diff
@@ -566,13 +583,38 @@ TDNFGetRepoMD(
                       pszTmpRepoDataDir,
                       TDNF_REPO_METADATA_FILE_NAME);
         BAIL_ON_TDNF_ERROR(dwError);
-
+        if(metalink)
+        {
+            dwError = TDNFAllocateStringPrintf(
```

Comment:
TDNF_SAFE_FREE_MEMORY(pszTmpRepoMetalinkFile) is missed in cleanup.

Reply:
Yeah, Thanks!